pcsx2/GS/Renderers/HW/GSRendererNew.cpp

```diff
@@ -567,14 +567,25 @@ void GSRendererNew::EmulateBlending(bool& DATE_PRIMID, bool& DATE_BARRIER)
 	// Blending doesn't require barrier, or sampling of the rt
 	const bool blend_non_recursive = !!(blend_flag & BLEND_NO_REC);
 
-	// BLEND MIX selection, use a mix of hw/sw blending
-	const bool blend_mix1 = !!(blend_flag & BLEND_MIX1);
-	const bool blend_mix2 = !!(blend_flag & BLEND_MIX2);
-	const bool blend_mix3 = !!(blend_flag & BLEND_MIX3);
-	bool blend_mix = (blend_mix1 || blend_mix2 || blend_mix3);
-
-	const bool alpha_c2_high_one = (ALPHA.C == 2 && ALPHA.FIX > 128u);
+	const bool alpha_c0_zero = (ALPHA.C == 0 && GetAlphaMinMax().max == 0);
+	const bool alpha_c0_one = (ALPHA.C == 0 && (GetAlphaMinMax().min == 128) && (GetAlphaMinMax().max == 128));
 	const bool alpha_c0_high_max_one = (ALPHA.C == 0 && GetAlphaMinMax().max > 128);
+	const bool alpha_c2_zero = (ALPHA.C == 2 && ALPHA.FIX == 0u);
+	const bool alpha_c2_one = (ALPHA.C == 2 && ALPHA.FIX == 128u);
+	const bool alpha_c2_high_one = (ALPHA.C == 2 && ALPHA.FIX > 128u);
+
+	// BLEND MIX HW/SW selection, use a mix of hw/sw blending
+	const bool blend_mix1_hw_sw = !!(blend_flag & BLEND_MIX1);
+	const bool blend_mix2_hw_sw = !!(blend_flag & BLEND_MIX2);
+	const bool blend_mix3_hw_sw = !!(blend_flag & BLEND_MIX3);
+	bool blend_mix_hw_sw = (blend_mix1_hw_sw || blend_mix2_hw_sw || blend_mix3_hw_sw);
+
+	// BLEND MIX SW selection, blend can be done in sw
+	const bool blend_mix1_sw = blend_mix1_hw_sw &&
+		((ALPHA.D == 2 && (alpha_c0_zero || alpha_c2_zero)) || (ALPHA.B == ALPHA.D && (alpha_c0_one || alpha_c2_one)));
+	const bool blend_mix2_sw = blend_mix2_hw_sw && (alpha_c0_zero || alpha_c2_zero);
+	const bool blend_mix3_sw = blend_mix3_hw_sw && (alpha_c0_zero || alpha_c2_zero);
+	const bool blend_mix_sw = (blend_mix1_sw || blend_mix2_sw || blend_mix3_sw);
```

Comment:
You could just do "if C is known to be zero, set A and B to 0" and it would solve pretty much all of this without needing 15 new flags

Reply:
Ah right, good idea.